lib/ret_web/schema/room_types.ex

```diff
@@ -0,0 +1,223 @@
+    field(:fly, :boolean)
+  end
+
+  @desc "Permissions for participants in the room"
+  object :member_permissions do
+    @desc "Allows non-admin participants to spawn and move media"
+    field(:spawn_and_move_media, :boolean)
+    @desc "Allows non-admin participants to spawn in-game cameras"
+    field(:spawn_camera, :boolean)
+    @desc "Allows non-admin participants to draw with a pen"
+    field(:spawn_drawing, :boolean)
+    @desc "Allows non-admin participants to pin media to the room"
+    field(:pin_objects, :boolean)
+    @desc "Allows non-admin participants to spawn emoji"
+    field(:spawn_emoji, :boolean)
+    @desc "Allows non-admin participants to toggle fly mode"
+    field(:fly, :boolean)
+  end
+
+  @desc "A room"
+  object :room do
+    @desc "The room's unique ID"
+    field(:hub_sid, :id, name: "id")
+    @desc "The room's name"
+    field(:name, :string)
+    @desc "The room's name as it appears at the end of its URL"
+    field(:slug, :string)
+    @desc "A description of the room"
+    field(:description, :string)
+    @desc "Makes this room public, while it is still open"
+    field(:allow_promotion, :boolean)
+    @desc "Temporary entry code"
+    field(:entry_code, :string)
+    @desc "Whether the room is open or closed"
+    field(:entry_mode, :string)
+    @desc "The host server associated with this room via the load balancer"
+    field(:host, :string)
+
+    @desc "The port number used to connect to the host server"
+    field(:port, :integer) do
+      @desc "The port number used to connect to the host server"
+      resolve(&Resolvers.RoomResolver.port/3)
+    end
+
+    @desc "TURN information for DLTS over TURN fallback, when enabled"
+    field(:turn, :turn_info) do
+      resolve(&Resolvers.RoomResolver.turn/3)
+    end
+
+    @desc """
+    Can be used to remove the X-Frame-Options header that is usually served to the Hubs client when this room is loaded, so that the client can access this room from a <frame>, <iframe>, <embed> or <object>.
+    """
+    field(:embed_token, :string) do
+      resolve(&Resolvers.RoomResolver.embed_token/3)
+    end
+
+    @desc "Permissions for participants in the room"
+    field(:member_permissions, :member_permissions) do
+      resolve(&Resolvers.RoomResolver.member_permissions/3)
+    end
+
+    @desc "Number of participants allowed to enter the room"
+    field(:room_size, :integer) do
+      resolve(&Resolvers.RoomResolver.room_size/3)
+    end
+
+    @desc "Number of participants in the room as avatars"
+    field(:member_count, :integer) do
+      resolve(&Resolvers.RoomResolver.member_count/3)
+    end
+
+    @desc "Number of participants in the room lobby"
+    field(:lobby_count, :integer) do
+      resolve(&Resolvers.RoomResolver.lobby_count/3)
+    end
+
+    @desc "Scene currently associated with the room"
+    field(:scene, :scene_or_scene_listing) do
+      resolve(dataloader(Scene))
+    end
+
+    # TODO: Figure out user_data
```

Comment:
I needed to add support for arbitrary json coming in and going out. Added that here https://github.com/mozilla/reticulum/pull/399/commits/e7aa1614b81a6742be22312a99b862afd1575244

Do you expect this to impact security at all? So far what I've considered is:
- [Risk] User sends huge payloads as `userData`. [Mitigation] We can cap request sizes like we do for file uploads.
- [Risk] User stores copyrighted or illegal content in `user_data`. [Mitigation] Take the same approach we do for file uploads.
- [Risk] User attempts a SQL injection attack or tries to store invalid json data. [Mitigation] I don't think the risk is any different here vs anywhere else we are parsing json. https://github.com/mozilla/reticulum/blob/feature/graphql-2/lib/ret_web/schema/types/custom/json.ex#L23

Reply:
This seems fine. I don't think there's any significant risk, since we're essentially treating this data as a black box and it's basically only for the user to produce and consume as they like. If we ever started parsing the data in the server, or even in our first-party client code, we'd have to be concerned, because at that point it becomes untrusted user input.
Putting a size limit on it seems like a good idea. Not sure if we can do that on the database column itself. Just some reasonably large upper bound. Doesn't have to be in this PR though.